### PR TITLE
Make traverse and sequence Fantasy-Land compliant

### PR DESCRIPTION
**Pull request by semmel**
### Motivation

- make Ramda work better with monad types provided by libraries like [Crocks](https://crocks.dev), [Sanctuary](https://github.com/sanctuary-js/sanctuary) and [fp/ts](https://gcanti.github.io/fp-ts/)
- #3337 , #3021 , #2235

### Changes to the implementations

This changes the way `sequence` and `traverse`'s implementations delegate to the Traversable's `.traverse` (i.e. `["fantasy-land/traverse"]`) if the data type is not `Array` but an Traversable `t`:
- `traverse` expects the Traversable `t` to conform to [Fantasy Land `traverse`](https://github.com/fantasyland/fantasy-land#fantasy-landtraverse-method), thus it no longer works with "old-style" Traversables like `ramda/test/shared/Id.js` (**Breaking change**) 
- `sequence` dispatches to `f.['fantasy-land/traverse'](F, x => x)` (or `f.traverse(F, x => x)`), thus also no longer works with "old-style" Traversables.

### Changes to the interface

`traverse` and `sequence` are extended to allow a [Type Representative](https://github.com/fantasyland/fantasy-land/tree/v4.0.1#type-representatives) `F` of the inner data type (the Applicative) as first argument.  Here it can be just a plain object with a single property `"fantasy-land/of"`.
Thus `traverse` and `sequence` now have the same signature as their [counterparts in Sanctuary](https://github.com/sanctuary-js/sanctuary#traverse--applicativeftraversablet--typerepf---a-fb---ta---ftb) and their [counterparts in Crocks](https://crocks.dev/docs/functions/pointfree-functions.html).

### CODE REVIEW

File: dist/ramda.js
Comment (kedashoe) [inline dist/ramda.js:1]: Could you revert these? We update these at release time

File: source/sequence.js [hunk 1]
```diff
@@ -26,16 +28,29 @@ import reduceRight from './reduceRight.js';
...
-      of([]),
-      traversable
-    );
+var sequence = _curry2(function sequence(F, traversable) {
+  const of = typeof F['fantasy-land/of'] === 'function'
+    ? F['fantasy-land/of']
+    : typeof F.of === 'function'
+      ? F.of
+      : F;
+  const TypeRep = { ['fantasy-land/of']: of };
```
Comment (kedashoe) [inline source/sequence.js:40]: Need brackets here?  `const TypeRep = { 'fantasy-land/of': of };`

File: source/sequence.js [hunk 2]
```diff
@@ -26,16 +28,29 @@ import reduceRight from './reduceRight.js';
...
+
+  return (
+    typeof traversable['fantasy-land/traverse'] === 'function'
+      ? traversable['fantasy-land/traverse'](TypeRep, x => x)
```
Comment (kedashoe) [inline source/sequence.js:44]: Same for arrow functions in source files. Could import `_identity`

**Review by kedashoe — CHANGES_REQUESTED** [review]
Looks great, ty! Just some cosmetic comments

**Comment by semmel** [comment]
@kedashoe Thanks for the kind review! 
I tried to address your points with the latest commits.

**Comment by kedashoe** [comment]
> @kedashoe Thanks for the kind review! I tried to address your points with the latest commits.

ty, looks good! I'll leave open for another day or two to allow others to comment before merging

**Comment by adispring** [comment]
Can we release a new version? There are so many changes and bugfixes merged into master since last release. @kedashoe @CrossEye

**Comment by kedashoe** [comment]
> Can we release a new version? There are so many changes and bugfixes merged into master since last release. @kedashoe @CrossEye

Been a while since I've done one. I'll give it a try this weekend.

**Pull request merged**